### PR TITLE
Fix Allocation `include` Filter Docs

### LINKED ISSUE #65113: index.routing.allocation.include  not work as document described
**Elasticsearch version** (`bin/elasticsearch --version`):
7.2.0

**Plugins installed**: []

**JVM version** (`java -version`):
java version "1.8.0_91"

**OS version** (`uname -a` if on a Unix-like system):
Linux   2.6.32-642.6.2.el6.x86_64 #1 SMP Wed Oct 26 06:52:09 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux

**Description of the problem including expected versus actual behavior**:

as the document say that different attributes for ``` index.routing.allocation.include.xxx ``` are all should satisfied
>If you specify multiple filters, all conditions must be satisfied for shards to be relocated. For example, to move the test index to big nodes in rack1, you could specify:

```
PUT test/_settings
{
  "index.routing.allocation.include.size": "big",
  "index.routing.allocation.include.rack": "rack1"
}
```
the document is [here](https://www.elastic.co/guide/en/elasticsearch/reference/7.2/shard-allocation-filtering.html#index-allocation-filters) 

but in my test i find that the test index will be relocated to nodes that have ever ``` big ```size OR  ```rack1``` attr

if i want to move the test index to big nodes in rack1, i should do like this 

```
PUT test/_settings
{
  "index.routing.allocation.require.size": "big",
  "index.routing.allocation.require.rack": "rack1"
}
```

### PR DESCRIPTION
**Pull request by original-brownbear**
Fix documentation to match actual behavior of `include` type filters.

Closes #65113

### CODE REVIEW

**Review by DaveCTurner — COMMENTED** [review]
The second example seems artificial to me, I think we can stick with the `.require.` one only.

Should we also spell out that if you have more than one class of filter (e.g. `include` _and_ `require`) then every class must be satisfied?

File: docs/reference/index-modules/allocation/filtering.asciidoc [hunk 1]
```diff
@@ -59,8 +59,21 @@ PUT test/_settings
 ------------------------
 // TEST[s/^/PUT test\n/]
 
-If you specify multiple filters, all conditions must be satisfied for shards to
-be relocated. For example, to move the `test` index to `big` nodes in `rack1`,
+If you specify multiple filters, all `require` type conditions and no `exclude` type conditions must be satisfied for shards to
```
Comment (original-brownbear) [inline docs/reference/index-modules/allocation/filtering.asciidoc:62]: Thanks @DaveCTurner makes sense about removing the second example. Maybe just replace this section by:

```
If you specify multiple filters the following conditions must be satisfied simultaneously by a node in order for shards to be relocated to it:

* If any `require` type conditions are specified, all of them must be satisfied
* If any `exclude` type conditions are specified, none of them may be satisfied
* If any `include` type conditions are specified, at least one of them must be satisfied

```

then, make the example below use `require` as I did here and revert the addition I made to `include` to keep it short and sweet?

Comment (DaveCTurner) [reply docs/reference/index-modules/allocation/filtering.asciidoc:62]: Sounds good to me (mod wrapping to 80 columns ofc)

Comment (original-brownbear) [reply docs/reference/index-modules/allocation/filtering.asciidoc]: Thanks, adjusted accordingly now :)

**Review by DaveCTurner — APPROVED** [review]
LGTM

**Comment by original-brownbear** [comment]
Thanks David!

**Pull request merged**